src/components/TodosList.tsx

```diff
@@ -0,0 +1,63 @@
+import { FC } from 'react';
+import { Todo } from '../types/Todo';
+import { TodoItem } from './TodoItem';
+import { useDispatch, useSelector } from '../providers/TodosContext';
+
+type Props = {
+  todos: Todo[]
+};
+
+export const TodosList: FC<Props> = ({ todos }) => {
+  const { tempTodo, inProcess } = useSelector();
+  const dispatch = useDispatch();
+
+  const handleDelete = (id: number) => {
+    dispatch({ type: 'setInProcess', payload: [...inProcess, id] });
+
+    const updatedTodos = todos.filter(todo => todo.id !== id);
+
+    dispatch({ type: 'setTodos', payload: updatedTodos });
+  };
```

Comment:
You should also modify this logic to send request to Api and delete the todo also there